Implement new stats/home page

selankon opened the pull request:
Fixes https://github.com/vocdoni/explorer/issues/77

Needs sdk: https://github.com/vocdoni/vocdoni-sdk/pull/406
Extended sdk: https://github.com/vocdoni/ui-components/pull/195

selankon committed:
ADd stats initial data

selankon committed:
Create icons map

selankon committed:
Improve stats cards

selankon committed:
Refactor blockCard to accept compact

selankon committed:
Pass grid props

selankon committed:
Refactor stats card

selankon committed:
Reuse component

selankon committed:
WIP Implement ChainCosts

selankon committed:
Use IncrementalStat

selankon committed:
Implement details grid info

selankon committed:
Implement show raw modal

selankon committed:
Create hint component

selankon committed:
Fix DetailsGrid layout

selankon committed:
Implement ChainCosts

selankon committed:
Minor layout fixes

selankon committed:
Mock txCosts

selankon committed:
Minor layout fixes

selankon committed:
Create wide layout

selankon committed:
Fix wide layout max width

selankon committed:
Fix dancing words

selankon committed:
Change read more

use an icon instead

selankon committed:
Fix image height

selankon committed:
Refactor to use right component

selankon committed:
Add close button to raw modal

selankon committed:
Refactor price factors

selankon committed:
Revert "Create wide layout"

This reverts commit 28fddc8b4fdbc3d7ef6a3408ed4c1ded79a7fe6b.

selankon committed:
Refactor chain info card

selankon committed:
Fix prettier

selankon committed:
Change homepage layout

selankon committed:
Add blockTimestamp

selankon committed:
Add blockTimestamp tooltip

selankon committed:
Multiple string fixes

selankon committed:
Show last 3 blocks

selankon committed:
Fix prettier

elboletaire reviewed (changes requested):
Without having checked the code much, some friendly reminders for when we're back of holidays:

- The tokens component takes too long to load, and there are no pending calls so it's probably something in the implementation
- We cannot publish that right column this empty... we should get the stats back to the right side, and maybe even put the tokens info inside the blockchain card
- @jpaulet said the `i` modal is not very intuitive, maybe he wants to propose something for that

elboletaire committed:
Update extended-sdk and remove mock

selankon committed:
Upgrade sdk commit

selankon committed:
Fix missing import

selankon commented:
> Without having checked the code much, some friendly reminders for when we're back of holidays:
> 
>     * The tokens component takes too long to load, and there are no pending calls so it's probably something in the implementation
> 
>     * We cannot publish that right column this empty... we should get the stats back to the right side, and maybe even put the tokens info inside the blockchain card
> 
>     * @jpaulet said the `i` modal is not very intuitive, maybe he wants to propose something for that

It seems that it was an error during the build so you was looking an outdated version of the stats page. 

Check it now because the tokens component have a better behavior now.

selankon committed:
Add error handling

It also refactors the `StatisticsCardWrapper` into a `StatsCardWrapper` component.

selankon committed:
Delete unused component

selankon committed:
Fix in sync badge color

selankon committed:
Use Raw IconButton

selankon committed:
Implement StatsModalWrapper

This way the price factors and VOC tokens price are shown inside two separated modals that can be opened from the Blockhain info card

elboletaire commented on src/components/Stats/ChainDetails/TxCosts.tsx:
```diff
@@ -0,0 +1,57 @@
+  }
+
+  return (
+    // <StatsCardWrapper
+    //   icon={Icons.TxIcon}
+    //   title={t('stats.voc_tokens', { defaultValue: 'VOC Tokens' })}
+    //   raw={txCosts}
+    //   isLoading={isLoading}
+    //   rightComp={<PriceFactorsModal />}
+    //   isError={isError}
+    //   error={error}
+    // >
+    <DetailsGrid rowGap={0} columnGap={6} details={prices} />
+    // </StatsCardWrapper>
```
Remove commented code please

selankon replied:
d2148c0

elboletaire commented on src/components/Stats/ChainDetails/StatsCards.tsx:
```diff
@@ -0,0 +1,164 @@
+  const [displayNumber, setDisplayNumber] = useState(0)
+  const duration = 2000
+
+  useEffect(() => {
```
Comment please

selankon replied:
d2148c0

elboletaire reviewed (changes requested):
Two minor issues, other than that LGTM

selankon committed:
Minor fixes

- Delete commented
- Comment useEffect
